Create stub files for new transformer using invoke script (#189)
MarketSquare/robotframework-tidy: CONTRIBUTING.md

Contribution guideline
======================

Getting started
----------------

> Did you spot bug or you really need some feature implemented? It's always best to start with create issue in github.

Reviewer: ```suggestion
Did you spot a bug or you really need some feature implemented? It's always best to start with creating issue in GitHub.
```

> It doesn't matter if you're willing to make change on your own. Creating issue allow us to discuss proposed changes (

Reviewer: ```suggestion
It doesn't matter if you're willing to make change on your own. Creating issue allows us to discuss proposed changes (
```

> and cover possible edge cases) and also prevents multiple people working on the same problem.

Reviewer: ```suggestion
and cover possible edge cases) and also prevents multiple people from working on the same problem.
```

You can also start discussion in our slack channel: [#tidy](https://robotframework.slack.com/archives/C01FR5992N6).

Setup
------
Create your own fork of Robotidy first (instructions [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo)). 
It will be easier to create Pull Requests later with your own fork. After cloning your fork to your disk install Robotidy
from the source code with `dev` profile. Run following in the root of the project:
```
pip install -e .[dev]
```
Option `-e` allows changes done by you to be automatically reloaded without need for reinstalling the package. `dev` profile 
installs extra libraries useful for testing and scripting (such as pytest or invoke).

Creating new transformer
------------------------

> When you want to add new transformer to Robotidy you can use `add-transformer` invoke helper script. It will create stub

Reviewer: ```suggestion
When you want to add new a transformer to Robotidy you can use `add-transformer` invoke helper script. It will create stub
```

> classes and tests. To do this run following command in project root:

Reviewer: ```suggestion
classes and tests. To do this run following command in the project root:
```

```
invoke add-transformer TRANSFORMER-NAME
```
It will create `TRANSFORMER-NAME.py` file in transformer directory with class stub and it will also create initial structure 
inside tests/atest/transformers directory. It will also add name of your transformer to `robotidy/transformers/__init__.py file` to 

> TRANSFORMERS lists. This list determine the order the transformers will be run. There are transformers that affect the others but if 

Reviewer: ```suggestion
TRANSFORMERS lists. This list determines the order of transformers that will be run. There are transformers that affect the others but if 
```

you're not sure how your transformer will work with other just leave the default order.

If you don't want to run your transformer with default ones (so it could be run only when selected with 

> `--transform` or configured with ``enabled=True`) add following flag to invoke script:

Reviewer: ```suggestion
`--transform` or configured with `enabled=True`) add the following flag to invoke script:
```

```
invoke add-transformer TRANSFORMER-NAME --disabled
```

Helper script will also add `docs/source/transformers/TRANSFORMER-NAME.rst` file. You can add full documentation 
of your transformer example together with more usage examples and possible configurations. Refer to other transformers documentation 
for used syntax (but your documentation engine use sphinx under the hood with alabaster theme and sphinx_tabs.tabs plugin). 

> Robotidy uses Robot Framework parsing api for visiting robot source files (parsed as ast trees). Visit [this](https://robot-framework.readthedocs.io/en/master/autodoc/robot.api.html#module-robot.api.parsing) 

Reviewer: ```suggestion
Robotidy uses Robot Framework Parsing API for visiting robot source files (parsed as AST trees). Visit [this](https://robot-framework.readthedocs.io/en/master/autodoc/robot.api.html#module-robot.api.parsing) 
```
